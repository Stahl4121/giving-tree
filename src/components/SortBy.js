import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import InputLabel from "@material-ui/core/InputLabel";
import MenuItem from "@material-ui/core/MenuItem";
import FormControl from "@material-ui/core/FormControl";
import Select from "@material-ui/core/Select";

const useStyles = makeStyles(theme => ({
  formControl: {
    minWidth: 120
  },
  selectEmpty: {
    marginTop: theme.spacing(2)
  }
}));

export default function SortBy(props) {
  const classes = useStyles();
  const [sort, setSort] = React.useState("");

  const inputLabel = React.useRef(null);
  const [labelWidth, setLabelWidth] = React.useState(0);
  React.useEffect(() => {
    setLabelWidth(inputLabel.current.offsetWidth);
  }, []);

  const handleChange = event => {
    setSort(event.target.value);
    props.parentCallback(event.target.value);
  };

  return (
    <div>
      <FormControl fullWidth variant="outlined" className={classes.formControl}>
        <InputLabel ref={inputLabel} id="select-radius-ref">
          Sort By
        </InputLabel>
        <Select
          labelId="select-sort-label"
          id="select-sort"
          value={sort}
          onChange={handleChange}
          labelWidth={labelWidth}
        >
          <MenuItem value="">
            <em>None</em>
          </MenuItem>
          <MenuItem value={"deadline"}>Deadline</MenuItem>
          <MenuItem value={"posting"}>Posting Date</MenuItem>
          <MenuItem value={"goalD"}>% of Goal Desc.</MenuItem>
          <MenuItem value={"goalI"}>% of Goal Inc.</MenuItem>
          <MenuItem value={"size"}>Goal Size</MenuItem>
        </Select>
      </FormControl>
    </div>
  );
}
