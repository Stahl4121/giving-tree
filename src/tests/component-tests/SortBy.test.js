import React from "react";
import { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import SortBy from "../../components/SortBy.js";

describe('Component: SortBy', () => {
  var enzyme = require('enzyme');
  enzyme.configure({ adapter: new Adapter() });

  it('Renders correctly', () => {
    const onSelect = jest.fn();
    const wrap = enzyme.mount(<SortBy parentCallback={onSelect} />)
    expect(wrap).toMatchSnapshot();
  });

  it('Sort by deadline', () => {
    const onSelect = jest.fn();
    const wrap = enzyme.mount(<SortBy parentCallback={onSelect} />)
    const select = wrap.find('#select-sort').at(0);
    select.props().onChange({ target: { value: "deadline"} });
    expect(onSelect.mock.calls.length).toBe(1);
    expect(onSelect.mock.calls[0][0]).toBe("deadline");
  });

  it('Sort by posting date', () => {
    const onSelect = jest.fn();
    const wrap = enzyme.mount(<SortBy parentCallback={onSelect} />)
    const select = wrap.find('#select-sort').at(0);
    select.props().onChange({ target: { value: "posting"} });
    expect(onSelect.mock.calls.length).toBe(1);
    expect(onSelect.mock.calls[0][0]).toBe("posting");
  });

  it('Sort by goal decreasing', () => {
    const onSelect = jest.fn();
    const wrap = enzyme.mount(<SortBy parentCallback={onSelect} />)
    const select = wrap.find('#select-sort').at(0);
    select.props().onChange({ target: { value: "goalD"} });
    expect(onSelect.mock.calls.length).toBe(1);
    expect(onSelect.mock.calls[0][0]).toBe("goalD");
  });

  it('Sort by goal increasing', () => {
    const onSelect = jest.fn();
    const wrap = enzyme.mount(<SortBy parentCallback={onSelect} />)
    const select = wrap.find('#select-sort').at(0);
    select.props().onChange({ target: { value: "goalI"} });
    expect(onSelect.mock.calls.length).toBe(1);
    expect(onSelect.mock.calls[0][0]).toBe("goalI");
  });

  it('Sort by size', () => {
    const onSelect = jest.fn();
    const wrap = enzyme.mount(<SortBy parentCallback={onSelect} />)
    const select = wrap.find('#select-sort').at(0);
    select.props().onChange({ target: { value: "size"} });
    expect(onSelect.mock.calls.length).toBe(1);
    expect(onSelect.mock.calls[0][0]).toBe("size");
  });

})