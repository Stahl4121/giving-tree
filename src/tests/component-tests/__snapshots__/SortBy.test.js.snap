// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SortBy Renders correctly 1`] = `ReactWrapper {}`;
